src/framework/ui/internal/navigationcontroller.cpp

```diff
@@ -820,6 +821,58 @@ void NavigationController::goToPrevPanel()
     goToPrevSection(true);
 }
 
+bool NavigationController::activateSearchControl(INavigationSection* activeSec)
+{
+    const std::string sectionName = activeSec->name().toStdString();
+    const std::string searchCtrl = "SearchControl";
+
+    for (const auto& panel : activeSec->panels()) {
+        if (panel->enabled()) {
+            if (requestActivateByName(sectionName, panel->name().toStdString(), searchCtrl)) {
+                INavigationControl* ctrl = activeControl();
+                doDeactivateControl(ctrl);
```

Comment:
Can you explain why it is necessary to deactivate the control that was just activated?